### PR TITLE
ddl: make output field name in `show tables/databases stmt` compatible with mysql

### LINKED ISSUE #35116: a little different in the field name with `SHOW TABLES LIKE` with MySQL and TiDB
## Enhancement

```
QUERY >	 SHOW TABLES LIKE 'apigw_routes'
TiDB  >
+-------------------+
| Tables_in_corteza |
+-------------------+
| apigw_routes      |
+-------------------+
MySQL >
+----------------------------------+
| Tables_in_corteza (apigw_routes) |
+----------------------------------+
| apigw_routes                     |
+----------------------------------+
```

### PR DESCRIPTION
**Pull request by likzn**
### What problem does this PR solve?

Issue Number: close #35116 

Problem Summary: `SHOW TABLES/DATABASES ... LIKE ...`,  the `output field` is a little different with Mysql.
```
QUERY >	 SHOW TABLES LIKE 'apigw_routes'
TiDB  >
+-------------------+
| Tables_in_corteza |
+-------------------+
| apigw_routes      |
+-------------------+
MySQL >
+----------------------------------+
| Tables_in_corteza (apigw_routes) |
+----------------------------------+
| apigw_routes                     |
+----------------------------------+
```
And the mysql will add `PatternLikeName` at the behind when we use `like`. If we do not use `like`, the behavior is the same.

Ref in Mysql: https://github.com/mysql/mysql-server/blob/3e90d07c3578e4da39dc1bce73559bbdf655c28c/sql/dd/info_schema/show.cc#L537-L545
### What is changed and how it works?
1. When we `buildShowSchema`, format field name with `PatternLikeExpr` to be compatible with Mysql.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Make output field name in statement of `show tables/databases` compatible with mysql
```

### CODE REVIEW

**Comment by likzn** [comment]
PTAL if free. By the way, how to join the slack of `tidb-community`? I input the email, but nothing response. And i also post in the `AskAug` https://asktug.com/t/topic/693265. But nobody reply yet

**Comment by xhebox** [comment]
> PTAL if free. By the way, how to join the slack of `tidb-community`? I input the email, but nothing response. And i also post in the `AskAug` https://asktug.com/t/topic/693265. But nobody reply yet

Should receive one email if you join the slack.

**Comment by likzn** [comment]
> > PTAL if free. By the way, how to join the slack of `tidb-community`? I input the email, but nothing response. And i also post in the `AskAug` https://asktug.com/t/topic/693265. But nobody reply yet
> 
> Should receive one email if you join the slack. Maybe in the trash.

It is strange. I had recived no email about slack..

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -4842,3 +4848,14 @@ func (b *PlanBuilder) buildCompactTable(node *ast.CompactTableStmt) (Plan, error
...
 }
+
+func extractPatternLikeName(patternLike *ast.PatternLikeExpr) string {
+	if patternLike == nil {
+		return ""
+	}
+	switch patternLike.Pattern.(type) {
+	case *driver.ValueExpr:
+		return patternLike.Pattern.(*driver.ValueExpr).GetString()
+	}
```
Comment (tangenta) [inline planner/core/planbuilder.go:4859]:
```suggestion
	switch v := patternLike.Pattern.(type) {
	case *driver.ValueExpr:
		return v.GetString()
	}
```

**Comment by likzn** [comment]
Has any friend take a look about this

File: planner/core/planbuilder.go [hunk 2]
```diff
@@ -2937,11 +2937,8 @@ func (b *PlanBuilder) buildShow(ctx context.Context, show *ast.ShowStmt) (Plan,
 		if (show.Tp == ast.ShowTables || show.Tp == ast.ShowTableStatus) && p.DBName == "" {
 			return nil, ErrNoDB
 		}
-		extractor := newShowBaseExtractor(*show)
-		if extractor.Extract() {
+		if extractor := newShowBaseExtractor(*show); extractor.Extract() {
 			p.Extractor = extractor
-			// Avoid building Selection.
```
[the remark this answers is not in the document]
Comment (xhebox) [reply planner/core/planbuilder.go:2943]: I think it is safer to mark an extra flag like `noShowPattern` or somethine similar.

**Review by xhebox — COMMENTED** [review]
LGTM except the removing of `show.Pattern = nil`

File: planner/core/planbuilder.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (likzn) [reply planner/core/planbuilder.go:2943]: Done. An extra flag is more clearer.

[the remark this answers is not in the document]
Comment (xhebox) [reply planner/core/planbuilder.go:2945]: > Done. An extra flag is more clearer.

You don't need to add it to `LogicalShow` struct, a variable in `buildShow` is enough :rofl:

[the remark this answers is not in the document]
Comment (likzn) [reply planner/core/planbuilder.go:2945]: Oh, yes. Just finished exercise, my head is a little dizzy

**Review by xhebox — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Comment by xhebox** [comment]
/hold

I think there is a stable test failure, @likzn , I'll take a look.

**Comment by likzn** [comment]
It is introduced by this PR. Error Message:
> show databases like \"test%\";: 
NEED: show tables like 't_';\nTables_in_lowercase_table\nT3\nt4\ndrop \n
BUT GOT: show tables like 't_';\nTables_in_lowercase_table (t_)\nT3\nt4\n\n"

The difference of them is the output field name. But i cant find where the `mysql_test` is. And i find test result in mysql is below https://github.com/mysql/mysql-server/blob/8d8c986e5716e38cb776b627a8eee9e92241b4ce/mysql-test/r/lowercase_table.result#L29-L32. 
image

**Pull request merged**